### ISSUE TITLE
MultiPage Tiff files - Suggestion

### ISSUE THREAD

**Issue by proudcanadianeh** [d2phap/ImageGlass#238]
Would love multi page tiff support. 

----
### Checklist
The below functions are applied to all `multiple pages/frames` that supported in ImageGlass, e.g. GIF, TIFF

- [x] Add navigation functions to main menu: `Main menu > Navigation > **`
- [x] Navigate image frames by hotkey: `Ctrl + <nav keys combo>`
  - [x] View next page
  - [x] View previous page
  - [x] View first page
  - [x] View last page
- [x] Display navigation controls as toolbox
  - [x] Create icons for View first/last page
- [x] Extract all image pages
- [x] Manage open state of the tool
- [x] Auto-show Page navigation tool for multi-page image
- [x] Localization
- [x] Add page index info to title bar

**Comment by RDev-AT** [comment]
For now this problem seems to be unsolved. Any progress since Q3 2017 so far?

**Comment by d2phap** [comment]
Hi @RDev-AT 
It's in the plan now: https://github.com/d2phap/ImageGlass/projects/5

**Comment by d2phap** [comment]
Hi, we are designing the viewing multi-page file function, do you have any suggestions how do you want to view it?

**Comment by proudcanadianeh** [comment]
I am going to reach out to an ex-coworker of mine and ask him to weigh in on this, as he uses tiffs extensively in his job and has tried out many different viewers

**Comment by proudcanadianeh** [comment]
So the feedback I got was that it is important to have something that visually distinguishes the file as being multi-page, and having controls to go through the pages that are visually distinct from the Next/Previous file arrows in the top bar. 
One idea was a translucent bar that appears on load of a multipage tiff that disappears automagically after 1.5 seconds, but reappears when moving the mouse near it again. Or perhaps a bar that appears near the bottom of the image with navigational info (Maybe thumbnails?). 

In summary, key things are to visually identify a file as being multipage in an obvious way and to have distinct navigational options that aren't to be confused with the standard next/previous photo buttons.

**Comment by d2phap** [comment]
Updated checklist for tracking

**Comment by d2phap** [comment]
I uploaded the first release for TIFF multi-page viewing, download it here: https://imageglass.org/moon

As this is a draft version, it's still being worked in progress (updating icons, fixing typos,...)

#### Navigation menu
![image](https://user-images.githubusercontent.com/3154213/67160109-f77be980-f37f-11e9-933b-fb8e55a4a987.png)

#### Extract image frames menu also extracts all TIFF pages
![image](https://user-images.githubusercontent.com/3154213/67160119-24300100-f380-11e9-9560-bcdae7d82203.png)

#### New Page nagivation tool in Tools menu
![image](https://user-images.githubusercontent.com/3154213/67160123-3ad65800-f380-11e9-8d67-dacea1e39dc9.png)


#### Page nagivation tool UI
![image](https://user-images.githubusercontent.com/3154213/67160130-56d9f980-f380-11e9-9763-1a0a13d73c1f.png)

**Comment by d2phap** [comment]
Updated: Add icons and language strings for page navigations
https://imageglass.org/moon

#### 🐛 Known bug: 
- The multipage file with different page size is not displayed correctly (i.g. `*.ICO` file)

![image](https://user-images.githubusercontent.com/3154213/67633541-ea18af00-f8eb-11e9-9598-60453ea75abe.png)

**Comment by d2phap** [comment]
- fixed the bug: The multipage file with different page size is not displayed correctly (i.g. *.ICO file)
- added page index info to title bar

You can try this feature in the latest build of ImageGlass Moon: https://imageglass.org/moon
